packages/indexer/src/services/priceFeeds.service.ts

```diff
@@ -0,0 +1,171 @@
+import { PriceUpdateFrequency, TOKEN_REGISTRY, TokenConfig } from "@/addresses/token-registry.addresses"
+import { PriceFeed, PriceFeedLog } from "@/configs/src/types"
+import { safeArray } from "@/utils/data.helper"
+import { normalizeTimestamp, timestampToDate } from "@/utils/date.helpers"
+import PriceHelper, { PriceResponse } from "@/utils/price.helpers"
+import stringify from "safe-stable-stringify"
+
+const PROVIDER_COINGECKO = "COINGECKO"
+
+/**
+ * Price Feeds Service fetches prices from CoinGecko adapter and stores them in the PriceFeed (current) and PriceFeedLog (historical).
+ */
+export class PriceFeedsService {
+	/**
+	 * getPrice is a static method that fetches the price of a token from CoinGecko and returns it as a PriceResponse object.
+	 * @param symbol - The symbol of the token to fetch the price for.
+	 * @param amount - The amount of the token to fetch the price for.
+	 * @param decimals - The number of decimals of the token to fetch the price for.
+	 * @returns A Promise that resolves to a PriceResponse object containing the price of the token in USD and the amount value in USD.
+	 * @dev This method trigger a background price fees update for all supported tokens
+	 */
+	static async getPrice(symbol: string, amount: bigint, decimals: number): Promise<PriceResponse> {
+		const priceFeed = await this.getLatestPrice(symbol)
```

Comment:
We should store token registry in db, whenever we try to fetch the price of a token that is not yet in our registry we should fetch the price from Coingecko and add the token  to the registry

Reply:
Don't add decimals. to the price feed, just fetch price and store directly, no conversions.